tagreader/clients.py

```diff
@@ -402,31 +407,41 @@ def get_units(self, tags):
             tags = [tags]
         units = {}
         for tag in tags:
-            if self.cache is not None:
-                r = self.cache.fetch_tag_metadata(tag, "unit")
-                if "unit" in r:
-                    units[tag] = r["unit"]
-            if tag not in units:
-                unit = self.handler._get_tag_unit(tag)
-                if self.cache is not None and unit is not None:
-                    self.cache.store_tag_metadata(tag, {"unit": unit})
-                units[tag] = unit
+            try:
+                if self.cache is not None:
+                    r = self.cache.fetch_tag_metadata(tag, "unit")
+                    if "unit" in r:
+                        units[tag] = r["unit"]
+                if tag not in units:
+                    unit = self.handler._get_tag_unit(tag)
+                    if self.cache is not None and unit is not None:
+                        self.cache.store_tag_metadata(tag, {"unit": unit})
+                    units[tag] = unit
+            except:
+                if self.search(tag) == []:  # check for nonexisting string
+                    print("Tag not found: " + str(tag))
```

Comment:
We should really have used a logger, but since we haven't configured one I suggest to do:
`warnings.warn(f"Tag not found: {tag}")`
Same for get_description()